src/frequenz/client/dispatch/_types.py

```diff
@@ -0,0 +1,348 @@
+
+    update_time: datetime
+    """The last update time of the dispatch in UTC. Set when a dispatch is modified."""
+
+    start_time: datetime
+    """The start time of the dispatch in UTC."""
+
+    duration: timedelta
+    """The duration of the dispatch, represented as a timedelta."""
+
+    selector: ComponentSelector
+    """The component selector specifying which components the dispatch targets."""
+
+    is_active: bool
+    """Indicates whether the dispatch is active and eligible for processing."""
+
+    is_dry_run: bool
+    """Indicates if the dispatch is a dry run.
+
+    Executed for logging and monitoring without affecting actual component states."""
+
+    # payload: Any  # Replace `Any` with your actual type for `google.protobuf.Struct`
+    # """The dispatch payload containing arbitrary data.
+
+    # It is structured as needed for the dispatch operation."""
+
+    recurrence: RecurrenceRule
+    """The recurrence rule for the dispatch.
+
+    Defining any repeating patterns or schedules."""
+
+    @classmethod
+    def from_protobuf(cls, pb_dispatch: PBDispatch) -> "Dispatch":
+        """Convert a protobuf dispatch to a dispatch.
+
+        Args:
+            pb_dispatch: The protobuf dispatch to convert.
+
+        Returns:
+            The converted dispatch.
+        """
+        return Dispatch(
+            id=pb_dispatch.id,
+            microgrid_id=pb_dispatch.microgrid_id,
+            type=pb_dispatch.type,
+            create_time=pb_dispatch.create_time.ToDatetime().replace(
+                tzinfo=timezone.utc
+            ),
+            update_time=pb_dispatch.update_time.ToDatetime().replace(
+                tzinfo=timezone.utc
+            ),
+            start_time=pb_dispatch.start_time.ToDatetime().replace(tzinfo=timezone.utc),
+            duration=timedelta(seconds=pb_dispatch.duration),
+            selector=_component_selector_from_protobuf(pb_dispatch.selector),
+            is_active=pb_dispatch.is_active,
+            is_dry_run=pb_dispatch.is_dry_run,
+            # payload=pb_dispatch.payload,
+            recurrence=RecurrenceRule.from_protobuf(pb_dispatch.recurrence),
+        )
+
+    def to_protobuf(self) -> PBDispatch:
+        """Convert a dispatch to a protobuf dispatch.
+
+        Returns:
+            The converted protobuf dispatch.
+        """
+        pb_dispatch = PBDispatch()
+
+        pb_dispatch.id = self.id
+        pb_dispatch.microgrid_id = self.microgrid_id
+        pb_dispatch.type = self.type
+        pb_dispatch.create_time.FromDatetime(self.create_time)
+        pb_dispatch.update_time.FromDatetime(self.update_time)
+        pb_dispatch.start_time.FromDatetime(self.start_time)
+        pb_dispatch.duration = int(self.duration.total_seconds())
+        pb_dispatch.selector.CopyFrom(_component_selector_to_protobuf(self.selector))
+        pb_dispatch.is_active = self.is_active
+        pb_dispatch.is_dry_run = self.is_dry_run
+        # pb_dispatch.payload = self.payload
```

Comment:
Why is this commented out?

Reply:
It caused trouble because the attribute is read-only. Haven'- investigated more

Reply:
Oh, weird. Maybe add a comment or create an issue about it?